### PR TITLE
Speed up findLocation in the absence of .debug_aranges

### PR DESCRIPTION
**Pull request by tudor**

### CODE REVIEW

**Comment by tudor** [comment]
The latest version does everything in one pass over the CU attributes.

File: folly/experimental/symbolizer/Dwarf.cpp [hunk 1]
```diff
@@ -477,41 +481,99 @@ bool Dwarf::findLocation(
...
+    kCompDir = 1U << 1,
+    kName = 1U << 2,
+    kLowPC = 1U << 3,
+    kHighPCOrRanges = 1U << 4,
+  };
+  unsigned expectedAttributes =
+    (kStmtList | kCompDir | kName | kLowPC | kHighPCOrRanges);
+  bool foundAddress = !checkAddress;
+  if (!foundAddress) {
+    expectedAttributes |= kHighPCOrRanges;
```
Comment (nbronson) [inline folly/experimental/symbolizer/Dwarf.cpp:497]: This bit is already set

File: folly/experimental/symbolizer/Dwarf.cpp [hunk 2]
```diff
@@ -477,41 +481,99 @@ bool Dwarf::findLocation(
...
+  bool seenHighPC = false;
+  enum : unsigned {
+    kStmtList = 1U << 0,
+    kCompDir = 1U << 1,
+    kName = 1U << 2,
+    kLowPC = 1U << 3,
+    kHighPCOrRanges = 1U << 4,
+  };
+  unsigned expectedAttributes =
+    (kStmtList | kCompDir | kName | kLowPC | kHighPCOrRanges);
```
Comment (nbronson) [inline folly/experimental/symbolizer/Dwarf.cpp]: kHighPCORanges isn't needed here

**Review by nbronson — APPROVED** [review]
LGTM, modulo the expectedAttributes initial value comment

File: folly/experimental/symbolizer/Dwarf.cpp (hunk 2 above)
Comment (tudor) [reply folly/experimental/symbolizer/Dwarf.cpp]: I thought I'd already fixed this. The intent was clearly there :)

**Pull request merged**